src/mapping/RadialBasisFctMapping.hpp

```diff
@@ -232,65 +266,122 @@ void RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::map(
   PRECICE_ASSERT(getDimensions() == output()->getDimensions(),
                  getDimensions(), output()->getDimensions());
 
-  Eigen::VectorXd &inValues  = input()->data(inputDataID)->values();
-  Eigen::VectorXd &outValues = output()->data(outputDataID)->values();
+  std::vector<double> inValues;
+  inValues.resize(input()->data(inputDataID)->values().size());
+  std::vector<double> outValues;
+  outValues.resize(output()->data(outputDataID)->values().size());
+
+  // Map the data from Eigen::VectorXd to std::vector for send and receive operations
+  Eigen::Map<Eigen::VectorXd>(&inValues[0], input()->data(inputDataID)->values().size())     = input()->data(inputDataID)->values();
+  Eigen::Map<Eigen::VectorXd>(&outValues[0], output()->data(outputDataID)->values().size())  = output()->data(outputDataID)->values();
+  
   int              valueDim  = input()->data(inputDataID)->getDimensions();
   PRECICE_ASSERT(valueDim == output()->data(outputDataID)->getDimensions(),
                  valueDim, output()->data(outputDataID)->getDimensions());
+  
   int deadDimensions = 0;
   for (int d = 0; d < getDimensions(); d++) {
     if (_deadAxis[d])
       deadDimensions += 1;
   }
   int polyparams = 1 + getDimensions() - deadDimensions;
 
-  if (getConstraint() == CONSERVATIVE) {
-    PRECICE_DEBUG("Map conservative");
-    static int      mappingIndex = 0;
-    Eigen::VectorXd Au(_matrixA.cols());  // rows == n
-    Eigen::VectorXd in(_matrixA.rows());  // rows == outputSize
-    Eigen::VectorXd out(_matrixA.cols()); // rows == n
-
-    // PRECICE_DEBUG("C rows=" << _matrixCLU.rows() << " cols=" << _matrixCLU.cols());
-    PRECICE_DEBUG("A rows=" << _matrixA.rows() << " cols=" << _matrixA.cols());
-    PRECICE_DEBUG("in size=" << in.size() << ", out size=" << out.size());
-
-    for (int dim = 0; dim < valueDim; dim++) {
-      for (int i = 0; i < in.size(); i++) { // Fill input data values
-        in[i] = inValues(i * valueDim + dim);
-      }
+  std::vector<double> globalInValues;
+  std::vector<double> globalOutValues;
 
-      Au  = _matrixA.transpose() * in;
-      out = _qr.solve(Au);
+  // Gather input data
+  if(utils::MasterSlave::isSlave()){
+    utils::MasterSlave::_communication->send(inValues, 0);
+  }
+  else {
+    globalInValues.resize(inValues.size());
+    globalInValues = inValues;
```

Comment:
This resizes the vector and then replaces it, thus you can omit the `resize()`.

If you know the total size beforehand then you can `u.reserve(n)` the memory in one go and insert the values from the other vector using `u.insert(u.end(), v.begin(), v.end())`.